monitor/src/test/java/edu/asu/diging/monitor/core/service/impl/AppManagerTest.java

```diff
@@ -21,84 +21,141 @@
     private String ID2 = "ID2";
     private AppTest test1;
     private String TEST1 = "TEST1";
-    
+
     @Before
     public void setUp() {
         MockitoAnnotations.initMocks(this);
-        
         app1 = new App();
         app1.setId(ID1);
         app2 = new App();
         app2.setId(ID2);
+        app3 = new App();
+        app3.setName("name");
+        app3.setUsername("");
+        app3.setPassword("");
         storedApps = new App[] { app1, app2 };
-        
+
         Mockito.when(dbConnection.getAllRegisteredApps()).thenReturn(storedApps);
         Mockito.when(dbConnection.getById(ID1)).thenReturn(app1);
-        
+        Mockito.when(dbConnection.update(app1)).thenReturn(app3);
+
         test1 = new AppTest();
         test1.setAppId(ID1);
         test1.setId(TEST1);
         app1.setLastTestId(TEST1);
         Mockito.when(appTestDbConnection.getById(TEST1)).thenReturn(test1);
     }
-    
+
     @Test
-    public void test_addApp_success() throws UnstorableObjectException, GroupNotFoundException {
+    public void test_addApp_with_encryption_success() throws UnstorableObjectException, GroupNotFoundException {
         String id = "ID1";
         Mockito.when(dbConnection.generateId()).thenReturn(id);
-        IApp app = managerToTest.addApp(new AppForm());
+        AppForm appForm = new AppForm();
+        appForm.setUsername("user");
+        appForm.setPassword("password");
+        IApp app = managerToTest.addApp(appForm);
+        Mockito.verify(passwordEncryptor).encrypt("password");
         Mockito.verify(dbConnection).store(app);
         Assert.assertEquals(id, app.getId());
     }
-    
+
+    @Test
+    public void test_addApp_without_encryption_success() throws UnstorableObjectException, GroupNotFoundException {
+        String id = "ID1";
+        Mockito.when(dbConnection.generateId()).thenReturn(id);
+        AppForm appForm = new AppForm();
+        appForm.setUsername("");
+        appForm.setPassword("");
+        IApp app = managerToTest.addApp(appForm);
+        Mockito.verify(passwordEncryptor, never()).encrypt("password");
+        Mockito.verify(dbConnection).store(app);
+        Assert.assertEquals(id, app.getId());
+    }
+
+    @Test
+    public void test_updateApp_with_encryption_success() throws UnstorableObjectException, GroupNotFoundException {
+        AppForm appForm = new AppForm();
+        appForm.setUsername("user");
+        appForm.setPassword("password");
+        app3.setUsername("user");
+        app3.setPassword("password");
+        appForm.setName(app3.getName());
+        IApp updatedApp = managerToTest.updateApp(app1, appForm);
+        Mockito.verify(passwordEncryptor).encrypt("password");
+        Mockito.verify(dbConnection).update(app1);
+        Assert.assertEquals(appForm.getName(), updatedApp.getName());
+        Assert.assertEquals(appForm.getUsername(), updatedApp.getUsername());
+        Assert.assertEquals(appForm.getPassword(), updatedApp.getPassword());
```

Comment:
that should not be the same, should it? after updating an app, the password should be encrypted, shouldn't it?